packages/idempotency/src/PersistenceLayerInterface.ts

```diff
@@ -0,0 +1,11 @@
+import { IdempotencyConfig } from './IdempotencyConfig';
+
+interface PersistenceLayerInterface {
+  configure(config: IdempotencyConfig): void
+  saveInProgress(): void
+  saveSuccess(): void
+  deleteRecord(): void
+  getRecord(): void
```

Comment:
I don't think we have discussed what a record looks like so I can make it at any type if you think that's more accurate. Some don't have parameters because we haven't gotten there. I can try to add some equivalents from the Python library and we can work them from there

Reply:
I have added the methods that the Python record uses, we'll have to discuss what we want out of the Python one...

this is what we currently have

`    def __init__(
        self,
        idempotency_key,
        status: str = "",
        expiry_timestamp: Optional[int] = None,
        in_progress_expiry_timestamp: Optional[int] = None,
        response_data: Optional[str] = "",
        payload_hash: Optional[str] = None,
    ) -> None:
        """

        Parameters
        ----------
        idempotency_key: str
            hashed representation of the idempotent data
        status: str, optional
            status of the idempotent record
        expiry_timestamp: int, optional
            time before the record should expire, in seconds
        in_progress_expiry_timestamp: int, optional
            time before the record should expire while in the INPROGRESS state, in seconds
        payload_hash: str, optional
            hashed representation of payload
        response_data: str, optional
            response data from previous executions using the record
        """
        self.idempotency_key = idempotency_key
        self.payload_hash = payload_hash
        self.expiry_timestamp = expiry_timestamp
        self.in_progress_expiry_timestamp = in_progress_expiry_timestamp
        self._status = status
        self.response_data = response_data`

Reply:
If you want to make things easier and prune as we go, I can just add them as properties and we can decide what we'll use later

Reply:
filled out a shell of record